tests/core/data_layer/test_data_store.py

```diff
@@ -1165,3 +1203,35 @@ async def test_data_server_files(data_store: DataStore, tree_id: bytes32, test_d
         current_root = await data_store.get_tree_root(tree_id=tree_id)
         assert current_root.node_hash == root.node_hash
         generation += 1
+
+
+@pytest.mark.asyncio
+async def test_pending_roots(data_store: DataStore, tree_id: bytes32) -> None:
+    key = b"\x01\x02"
+    value = b"abc"
+
+    await data_store.insert(
+        key=key,
+        value=value,
+        tree_id=tree_id,
+        reference_node_hash=None,
+        side=None,
+        status=Status.COMMITTED,
+    )
+
+    key = b"\x01\x03"
+    value = b"abc"
+
+    await data_store.autoinsert(
+        key=key,
+        value=value,
+        tree_id=tree_id,
+        status=Status.PENDING,
+    )
+    pending_root = await data_store.get_pending_root(tree_id=tree_id)
+    assert pending_root is not None
+    assert pending_root.generation == 2 and pending_root.status == Status.PENDING
+
+    await data_store.clear_pending_roots(tree_id=tree_id)
+    pending_root = await data_store.get_pending_root(tree_id=tree_id)
+    assert pending_root is None
```

Comment:
Could assert we're back at the expected not-pending root instead of having accidentally confirmed it or such.